guides/documentations/essentials.md

```diff
@@ -0,0 +1,374 @@
+# How Plugin-core works
+
+Leto-modelizer is a tool that allows you to create your own infrastructure code via diagram by using a graphical interface. Behind this interface, the solution writes the code equivalent to the selected bricks (called Components) in an infrastructure language as code chosen via a plugin (Terraform, Ansible, Kubernetes resources, etc.).
+
+| Examples of diagrams |
+| :--: |
+| <img height="400" src="../../documentations/diagrams.png"/> |
+
+Plugin-core is build to contain all the base code needed for modeling tools to create those diagrams in Leto's projects. You can use this library to create your own plugin, by overriding classes and methods.
+
+Here are all classes you can use in your plugin:
+
+```js
+DefaultMetadata.js,              // Class that represents the metadata of a specific declarative code implementation.
+DefaultRender.js,                // Class that compiles a Component to data.
+DefaultParser.js,                // Class used for parsing data to Component.
```

Comment:
Class used for VERB + [-ing] OR Class that VERBs => choose one

Reply:
too restrictive